app/services/orphaned_profiles_cleaner.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+# Service for cleaning up orphaned profiles
+class OrphanedProfilesCleaner
+  class << self
+    def run!
+      # No need to run this in an empty DB
+      return if Profile.canonical(false).none?
+
+      reassign_by_parent
+      Profile.canonical(false).where(policy_id: nil).destroy_all # Remove remaining orphaned profiles
+    end
+
+    private
+
+    def reassign_by_parent
+      ActiveRecord::Base.connection.execute(%(
+        UPDATE "profiles" SET "policy_id" = "sq"."policy_id" FROM (
+          SELECT "profiles"."id", "xp"."policy_id" FROM "profiles"
+          INNER JOIN "profiles" as "xp" ON "xp"."parent_profile_id" = "profiles"."parent_profile_id"
```

Comment:
Could this create duplicate profiles of same parent hence same benchmark being assigned to a policy?

Reply:
Umm, I don't understand the question...

Reply:
Resolved in private chat. The duplicates might exist, however they could differ by OS minor version (which might not be the case for our db).